cwm_worker_tests/upload_download_test.py

```diff
@@ -47,16 +47,50 @@ def upload(endpoint, access_key, secret_key, bucket, num_files, file_size, outpu
         # Create test file, remove at the end
         filepath = create_test_file(file_size)
 
+        uploads = []
+
         print(f'Uploading files...')
         for i in range(num_files):
+            file_upload_start_time = datetime.datetime.now()
             filename = get_upload_filename()
-            client.fput_object(bucket, filename, filepath)
+            error = None
+            try:
+                client.fput_object(bucket, filename, filepath)
+            except S3Error as e:
+                error = str(e)
+            except Exception as e:
+                error = str(e)
+            file_upload_end_time = datetime.datetime.now()
+            file_upload_elapsed_time_seconds = (file_upload_end_time - file_upload_start_time).total_seconds()
             print(f'\rUploaded: {i+1} files', end='', flush=True)
+            uploads.append({
```

Comment:
all uploads should succeed, otherwise test will be with invalid number of files

Reply:
What if an upload fails? Treat it as a fatal error and stop at once?

Reply:
yes